sycl/unittests/pi/PlatformTest.cpp

```diff
@@ -6,80 +6,87 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include <CL/sycl.hpp>
 #include <CL/sycl/detail/pi.hpp>
 #include <gtest/gtest.h>
-#include <memory>
+#include <vector>
+
+namespace {
 
 using namespace cl::sycl;
 
-namespace pi {
 class PlatformTest : public ::testing::Test {
 protected:
+  std::vector<pi_platform> _platforms;
 
-constexpr static size_t out_string_size =
-    8192u; // Using values from OpenCL CTS clGetPlatforms test
+  PlatformTest() : _platforms{} { detail::pi::initialize(); };
 
-  PlatformTest() { detail::pi::initialize(); }
+  ~PlatformTest() override = default;
 
-  ~PlatformTest() = default;
-};
+  void SetUp() {
+    ASSERT_NO_FATAL_FAILURE(Test::SetUp());
 
-TEST_F(PlatformTest, piPlatformsGet) {
-  pi_uint32 platformCount = 0;
+    const static char *platform_count_key = "PiPlatformCount";
```

Comment:
I am not following the trick with the platform_count_key & RecordProperty. Could you explain how this is helpful?

Reply:
Gtest can output log files from the tests (`--gtest_output=xml:/path/to/log/file.xml` option). The `RecordProperty` function adds extra key-value pairs into this output. Later logs to the same key overwrite earlier values on output of the file.

To simplify parsing of the output file I try to always have the key `PiPlatformCount` in it: first it is set to `0` so even if the following assertion fails, it is in the log. If the assertion does not fail then I overwrite it with the actual value queried.

Reply:
Should the log ever be used, then this should allow checking the log for platform counts instead of needing to trawl the warning message from the output stream.

Reply:
Cool, thanks for the explanation.